zigpy/backports/enum.py

```diff
@@ -0,0 +1,35 @@
+"""Enum backports from standard lib."""
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, TypeVar
+
+_StrEnumSelfT = TypeVar("_StrEnumSelfT", bound="StrEnum")
+
+
+class StrEnum(str, Enum):
+    """Partial backport of Python 3.11's StrEnum for our basic use cases."""
+
+    def __new__(
+        cls: type[_StrEnumSelfT], value: str, *args: Any, **kwargs: Any
+    ) -> _StrEnumSelfT:  # noqa: PYI019
```

Comment:
`# noqa: PYI019` is needed for ruff here, as it'll otherwise complain about this:
``PYI019 Methods like `__new__` should return `typing.Self` instead of a custom `TypeVar``
... which we can't use because `typing.Self` was only added in Python 3.11 and we still support 3.9, 3.10, and 3.10.